pkg/kvstore/etcd.go

```diff
@@ -999,7 +1165,30 @@ func (e *etcdClient) CreateIfExists(condKey, key string, value []byte, lease boo
 
 // ListPrefixLocked returns a list of keys matching the prefix only if the client is still holding the given lock.
 func (e *etcdClient) ListPrefixLocked(prefix string, lock KVLocker) (KeyValuePairs, error) {
-	return e.ListPrefix(prefix)
+	duration := spanstat.Start()
+	e.limiter.Wait(ctx.TODO())
+	opGet := client.OpGet(prefix, client.WithPrefix())
+	cmp := lock.Comparator().(client.Cmp)
+	txnReply, err := e.client.Txn(context.Background()).If(cmp).Then(opGet).Commit()
```

Comment:
Is the error always guaranteed to be nil in cases were `If(cmp)` fails? I see that `txnReply` has the following documentation for its `Succeeded` field, which is in line with what you are doing:

```
// succeeded is set to true if the compare evaluated to true or false otherwise.
```

Reply:
>Is the error always guaranteed to be nil in cases were If(cmp) fails?

Yes.